.github/workflows/main.yml

```diff
@@ -23,3 +23,20 @@ jobs:
           SURGE_TOKEN: 4f75270ec1d0e4d48c5ecc3959f9d888
           DEPLOY_DOMAIN: https://pr-${{ github.event.number }}-sef-site.surge.sh
         run: surge ./ $DEPLOY_DOMAIN --token $SURGE_TOKEN
+      - name: Comment on the PR with the preview link
+        uses: actions/github-script@v5
+        env:
+          BODY: "Hey @${{ github.event.pull_request.user.login }}! 👋\n 
+              Here's the live preview link for your PR!\n
+              https://pr-${{ github.event.number }}-sef-site.surge.sh \n\n
+              Thank you very much for opening a pull request at SEF!\n
+              Every little contribution you make, helps us to improve our home crafted instruments that aid the process of making the current education system more sustainable." 
+        with:
+          github-token: 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
+          script: |
+            github.rest.issues.createComment({
+              issue_number: context.issue.number,
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              body: "test"
```

Comment:
Is this correct?

Reply:
No these are some test commits. The correct commit is https://github.com/sef-global/sef-site/pull/1158/commits/5ed62d66ff25d594f69fb32b9192cdad7b5e63cb

I'll work this further and let you know.

Reply:
It seems it's impossible to achieve this without exposing a token. One solution might be calling the github API from using our Leaderboard.